src/ui/public/field_editor/field_editor.js

```diff
@@ -378,19 +432,28 @@ export class FieldEditor extends PureComponent {
     return this.state.showDeleteModal ? (
       <EuiOverlayMask>
         <EuiConfirmModal
-          title={`Delete field '${field.name}'`}
+          title={this.props.intl.formatMessage(
+            { id: 'common.ui.fieldEditor.deleteField.header', defaultMessage: 'Delete field \'{fieldName}\'' },
+            { fieldName: field.name })}
           onCancel={this.hideDeleteModal}
           onConfirm={() => {
             this.hideDeleteModal();
-            this.deleteField();
+            this.deleteField(this.props.intl);
           }}
-          cancelButtonText="Cancel"
-          confirmButtonText="Delete"
+          cancelButtonText={this.props.intl.formatMessage(
+            { id: 'common.ui.fieldEditor.deleteField.cancel.button', defaultMessage: 'Cancel' })}
+          confirmButtonText={this.props.intl.formatMessage(
+            { id: 'common.ui.fieldEditor.deleteField.delete.button', defaultMessage: 'Delete' })}
           buttonColor="danger"
           defaultFocusedButton={EUI_MODAL_CONFIRM_BUTTON}
         >
-          <p>You can&apos;t recover a deleted field.</p>
-          <p>Are you sure you want to do this?</p>
+          <p>
+            <FormattedMessage
+              id="common.ui.fieldEditor.deleteField.label"
+              defaultMessage="You can't recover a deleted field.{newLine}Are you sure you want to do this?"
```

Comment:
Technically this results in invalid markup (paragraphs inside of spans). Can we render each line within its own paragraph, as it was originally?

![image](https://user-images.githubusercontent.com/1238659/42788699-e0e748e8-8915-11e8-897b-ad513b5455e9.png)

Reply:
The main concern at the splitting paragraphs here is that we loose the context for the second paragraph. I mean that when we detach `Are you sure you want to do this?` from the first paragraph we loose the meaning of `this`. It's important for translation vendor.

As an option I applied `<br/>` as paragraphs separator.